pkg/manager/member/pvc_resizer.go

```diff
@@ -385,132 +375,284 @@ func (p *pvcResizer) updateVolumeStatus(ctx *componentVolumeContext) {
+				pvcID, cid, *pvc.Spec.StorageClassName)
+		}
+
+		needResizeVolumes = append(needResizeVolumes, volume)
+	}
+
+	return map[volumePhase][]*volume{
+		needResize: needResizeVolumes,
+		resizing:   resizingVolumes,
+		resized:    resizedVolumes,
+	}, nil
+}
+
+func (p *pvcResizer) resizeVolumesForPod(ctx *componentVolumeContext, pod *corev1.Pod, volumes []*volume) error {
+	if err := p.beforeResizeForPod(ctx, pod, volumes); err != nil {
+		return err
+	}
 
-			allResized = false
+	desiredVolumeQuantity := ctx.desiredVolumeQuantity
+	errs := []error{}
+
+	for _, volume := range volumes {
+		volName := volume.name
+		pvc := volume.pvc
+		pvcID := fmt.Sprintf("%s/%s", pvc.Namespace, pvc.Name)
+
+		currentRequest, ok := pvc.Spec.Resources.Requests[corev1.ResourceStorage]
+		if !ok {
+			errs = append(errs, fmt.Errorf("resize PVC %s failed: storage request is empty", pvcID))
+			continue
+		}
+		quantityInSpec, exist := desiredVolumeQuantity[volName]
+		if !exist {
+			errs = append(errs, fmt.Errorf("resize PVC %s failed: not exist in desired volumes", pvcID))
+			continue
+		}
 
-			// patch PVC to expand the storage
-			mergePatch, err := json.Marshal(map[string]interface{}{
-				"spec": map[string]interface{}{
-					"resources": corev1.ResourceRequirements{
-						Requests: corev1.ResourceList{
-							corev1.ResourceStorage: quantityInSpec,
-						},
+		mergePatch, err := json.Marshal(map[string]interface{}{
+			"spec": map[string]interface{}{
+				"resources": corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceStorage: quantityInSpec,
 					},
 				},
-			})
-			if err != nil {
-				errors = append(errors, err)
-				continue
+			},
+		})
+		if err != nil {
+			errs = append(errs, fmt.Errorf("resize PVC %s failed: %s", pvcID, err))
+			continue
+		}
+		_, err = p.deps.KubeClientset.CoreV1().PersistentVolumeClaims(pvc.Namespace).Patch(context.TODO(), pvc.Name, types.MergePatchType, mergePatch, metav1.PatchOptions{})
+		if err != nil {
+			errs = append(errs, fmt.Errorf("resize PVC %s failed: %s", pvcID, err))
+			continue
+		}
+
+		klog.Infof("resize PVC %s of %s: storage request is updated from %s to %s",
+			pvcID, ctx.ComponentID(), currentRequest.String(), quantityInSpec.String())
+	}
+
+	return errutil.NewAggregate(errs)
+}
+
+func (p *pvcResizer) beforeResizeForPod(ctx *componentVolumeContext, resizePod *corev1.Pod, volumes []*volume) error {
+	logPrefix := fmt.Sprintf("before resizing volumes of Pod %s/%s for %q", resizePod.Namespace, resizePod.Name, ctx.ComponentID())
+
+	switch ctx.status.GetMemberType() {
+	case v1alpha1.TiKVMemberType:
+		tc, ok := ctx.cluster.(*v1alpha1.TidbCluster)
+		if !ok {
+			return fmt.Errorf("%s: cluster is not tidb cluster", logPrefix)
+		}
+
+		if !tc.TiKVAllStoresReady() {
+			return fmt.Errorf("%s: any store is not ready", logPrefix)
+		}
+
+		// remove leader eviction ann from reized pod
+		for _, podVolume := range ctx.actualPodVolumes {
```

Comment:
Why remove the anno before resizing? We should do that after the resizing is done, right?

Reply:
Is it more readable that we check the store status UP and remove the annotation before we decide to scale PVCs for another Pod in L400 block?